lib/bolt/config.rb

```diff
@@ -593,8 +651,9 @@ def plugins
     def plugin_hooks
       if @data['plugin-hooks'].any? && @data['plugin_hooks'].any?
         Bolt::Logger.warn_once(
-          "plugin-hooks and plugin_hooks set",
-          "Detected configuration for 'plugin-hooks' and 'plugin_hooks'. Bolt will ignore 'plugin_hooks'."
+          id:  "plugin_hooks_conflict",
+          msg: "Detected configuration for 'plugin-hooks' and 'plugin_hooks'. Bolt will ignore 'plugin_hooks'.",
+          src: self
```

Comment:
Definitely wish there was a better way to just get the calling class from the method instead of having to do this *everywhere* :P

Reply:
Ya :(

Reply:
Would it be cleaner to just extend to `Logging::Logger` class with a `warn_once` method? That does make sharing the cache a little tricky though...